src/mapDataManipulation/effectFunctions.ts

```diff
@@ -66,3 +66,28 @@ export function runColorEffectFunction(
 
   return rgbMappingArrays;
 }
+
+export function runCustomEffectFunctions(
+  rgbMappingArrays: RgbMappingArrays,
+  effects: Effects,
+): RgbMappingArrays {
+  if (!isEffectSet(effects.customEffect)) {
+    return rgbMappingArrays;
+  }
+
+  let newRgbMappingArrays = { ...rgbMappingArrays };
```

Comment:
IIUC, this is not needed? It might also be a bit misleading, because even if we are copying the parent object, the values are actually being changed in-place. I would suggest simply using `rgbMappingArrays` directly, to explicitly show that we are mutating the structure.